src/algebra/jordan/basic.lean

```diff
@@ -0,0 +1,262 @@
+([jordanvonneumannwigner1934]) as a mathematical model for the observables of a quantum mechanical
+physical system (for a C*-algebra the self-adjoint part is closed under the symmetrised Jordan
+multiplication). Jordan algebras have subsequently been studied from the points of view of abstract
+algebra and functional analysis. They have connections to Lie algebras and differential geometry.
+
+A more general concept of a (non-commutative) Jordan algebra can also be defined, as a
+(non-commutative, non-associative) algebra `A` where, for each `a` in `A`, the operators of left and
+right multiplication by `a` and `a^2` commute. Such algebras have connected to the Vidav-Palmer
+theorem [cabreragarciarodriguezpalacios2014].
+
+A comprehensive overview of the algebraic theory can be found in [mccrimmon2004].
+
+A real Jordan algebra `A` can be introduced by
+```
+variables {A : Type*} [non_unital_non_assoc_ring A] [module ℝ A] [smul_comm_class ℝ A A]
+  [is_scalar_tower ℝ A A] [comm_jordan A]
+```
+
+## Main results
+
+- `lin_jordan` : Linearisation of the commutative Jordan axiom
+
+## Implementation notes
+
+We shall primarily be interested in linear Jordan algebras (i.e. over rings of characteristic not
+two) leaving quadratic algebras to those better versed in that theory.
+
+The conventional way to linearise the Jordan axiom is to equate coefficients (more formally, assume
+that the axiom holds in all field extensions). For simplicity we use brute force algebraic expansion
+and substitution instead.
+
+## References
+
+* [Cabrera García and Rodríguez Palacios, Non-associative normed algebras. Volume 1]
+  [cabreragarciarodriguezpalacios2014]
+* [Hanche-Olsen and Størmer, Jordan Operator Algebras][hancheolsenstormer1984]
+* [Jordan, von Neumann and Wigner, 1934][jordanvonneumannwigner1934]
+* [McCrimmon, A taste of Jordan algebras][mccrimmon2004]
+
+
+-/
+
+set_option old_structure_cmd true
+
+section non_unital_non_assoc_ring
+
+variables {A : Type*} [non_unital_non_assoc_semiring A]
+
+namespace non_unital_non_assoc_ring
+/--
+Left multiplication operator
+-/
+@[simps] def L : A →+ add_monoid.End A := add_monoid_hom.mul
+
+/--
+Right multiplication operator
+-/
+@[simps] def R : A→+(add_monoid.End A) := add_monoid_hom.flip (L : A →+ add_monoid.End A)
+
+lemma L_def (a b : A) : L a b = a*b := rfl
+
+lemma R_def (a b : A) : R a b = b*a := rfl
+
+end non_unital_non_assoc_ring
+
+end non_unital_non_assoc_ring
+
+open non_unital_non_assoc_ring
+/--
+A non unital, non-associative ring with a (non-commutative) Jordan multiplication.
+-/
+class jordan (A : Type*) [non_unital_non_assoc_ring A] :=
+(commL1R1: ∀ a b : A, (a * b) * a = a * (b * a))
+(commL1L2: ∀ a b : A, a * ((a * a) * b) = (a * a) * (a * b))
+(commL1R2: ∀ a b : A, a* (b * (a * a)) = (a * b) * (a * a) )
+(commL2R1: ∀ a b : A, (a * a) * (b * a) = ((a * a) * b) * a)
+(commR1R2: ∀ a b : A, (b * (a * a)) * a = (b * a) * (a * a))
```

Comment:
I think we could improve the naming here. How about:
```suggestion
class jordan_algebra (A : Type*) [non_unital_non_assoc_ring A] :=
(lmul_comm_rmul : ∀ a b : A, (a * b) * a = a * (b * a))
(lmul_lmul_comm_lmul: ∀ a b : A, (a * a) * (a * b) = a * ((a * a) * b))
(lmul_lmul_comm_rmul: ∀ a b : A, (a * a) * (b * a) = ((a * a) * b) * a)
(lmul_comm_rmul_rmul: ∀ a b : A, (a * b) * (a * a) = a * (b * (a * a)))
(rmul_comm_rmul_rmul: ∀ a b : A, (b * a) * (a * a) = (b * (a * a)) * a)
```

Reply:
@ocfnash  - should this be `jordan_ring` rather than `jordan_algebra` (since we aren't specifying bilinearity over ring of scalars)?

Reply:
Oops, yes `jordan_ring`.

Reply:
And presumably `comm_jordan` -> `comm_jordan_ring` as well.